File: src/test/java/org/veupathdb/lib/blast/field/DustTest.java
package org.veupathdb.lib.blast.field;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

@DisplayName("Dust")
class DustTest
{
  private final ObjectMapper json = new ObjectMapper();

  @Test
  @DisplayName("\"yes\" dust serialization")
  void yesDustOut() throws Exception {
    var tgt = Dust.yesDust();
    assertEquals("\"yes\"", json.writeValueAsString(tgt));
  }

  @Test
  @DisplayName("\"yes\" dust deserialization")
  void yesDustIn() throws Exception {
    var raw = "\"yes\"";
    var tgt = json.readValue(raw, Dust.class);
    assertTrue(tgt.isYes());
  }

  @Test
  @DisplayName("\"no\" dust serialization")
  void noDustOut() throws Exception {
    var tgt = Dust.noDust();
    assertEquals("\"no\"", json.writeValueAsString(tgt));
  }

  @Test
  @DisplayName("\"no\" dust deserialization")
  void noDustIn() throws Exception {
    var raw = "\"no\"";
    var tgt = json.readValue(raw, Dust.class);
    assertTrue(tgt.isNo());
  }

  @Test
  @DisplayName("\"lwl\" dust serialization")
  void lwlDustOut() throws Exception {
    var tgt = Dust.lwlDust(10, 12, 14);
    assertEquals("{\"level\":10,\"window\":12,\"linker\":14}", json.writeValueAsString(tgt));
  }

  @Test
  @DisplayName("\"lwl\" dust deserialization")
  void lwlDustIn() throws Exception {
    var raw = "{\"level\": 10,\"window\": 12, \"linker\": 14}";
    var tgt = json.readValue(raw, Dust.class);
    assertEquals(10, tgt.getLevel());
    assertEquals(12, tgt.getWindow());
    assertEquals(14, tgt.getLinker());
  }
}

File: src/test/java/org/veupathdb/lib/blast/field/DustTest.kt
package org.veupathdb.lib.blast.field

import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.veupathdb.lib.blast.field.Dust.Companion.lwlDust
import org.veupathdb.lib.blast.field.Dust.Companion.noDust
import org.veupathdb.lib.blast.field.Dust.Companion.yesDust

@DisplayName("Dust")
internal class DustTest {
  private val json = ObjectMapper()
  @Test
  @DisplayName("\"yes\" dust serialization")
  @Throws(Exception::class)
  fun yesDustOut() {
    val tgt = yesDust()
    Assertions.assertEquals("\"yes\"", json.writeValueAsString(tgt))
  }

  @Test
  @DisplayName("\"yes\" dust deserialization")
  @Throws(Exception::class)
  fun yesDustIn() {
    val raw = "\"yes\""
    val tgt = json.readValue(raw, Dust::class.java)
    Assertions.assertTrue(tgt.isYes)
  }

  @Test
  @DisplayName("\"no\" dust serialization")
  @Throws(Exception::class)
  fun noDustOut() {
    val tgt = noDust()
    Assertions.assertEquals("\"no\"", json.writeValueAsString(tgt))
  }

  @Test
  @DisplayName("\"no\" dust deserialization")
  @Throws(Exception::class)
  fun noDustIn() {
    val raw = "\"no\""
    val tgt = json.readValue(raw, Dust::class.java)
    Assertions.assertTrue(tgt.isNo)
  }

  @Test
  @DisplayName("\"lwl\" dust serialization")
  @Throws(Exception::class)
  fun lwlDustOut() {
    val tgt = lwlDust(10, 12, 14)
    Assertions.assertEquals("{\"level\":10,\"window\":12,\"linker\":14}", json.writeValueAsString(tgt))
  }

  @Test
  @DisplayName("\"lwl\" dust deserialization")
  @Throws(Exception::class)
  fun lwlDustIn() {
    val raw = "{\"level\": 10,\"window\": 12, \"linker\": 14}"
    val tgt = json.readValue(raw, Dust::class.java)
    Assertions.assertEquals(10, tgt.level)
    Assertions.assertEquals(12, tgt.window)
    Assertions.assertEquals(14, tgt.linker)
  }
}